src/app/js/public/selectors.js

```diff
@@ -40,6 +41,10 @@ export const fromSearch = createGlobalSelectors(
     localFromSearch,
 );
 
+const getFacet = page => state => state[page].facet;
+export const fromFacet = page =>
```

Comment:
What about using a dictionary.
```js
facetDictionary = {
    search:  createGlobalSelectors(getFacet('search'), localFromFacet),
    ...
}

fromFacet = page => 
    facetDictionary[page];
```
This way the selector will not be recreated each time and you will benefits from reselect memoization.